### PR TITLE
Brave Actions sized at 18px and highlighted correctly

### PR DESCRIPTION
**Pull request by petemill**
**Updates brave-extension dep - here's what's in that update: https://github.com/brave/brave-extension/compare/038a43af5294150d48ce94a326290f70aa001cd0...70550a1619139459119f472c18a03dc12c3ed5f2**

Fix brave/brave-browser#2295
Fix brave/brave-browser#2807

- Increases size of icons
- Custom drawing for buttons' highlight background so that right side is inset, including a right margin within the button itself. The badge can therefore be drawn within this area if it gets too wide.
- Anchors the badge 12px from right side of highlight / 14px from right side of entire control.
- Re-centering of icon within custom highlight inset and not within the button rectangle that contains the right inset.

image

image

image

image

image

image

image

image

image

image

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Windows
  - [x] macOS
  - [ ] Linux
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on
  - [ ] Windows
  - [ ] macOS
  - [ ] Linux
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [ ] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone

## Test Plan:
See https://github.com/brave/brave-browser/issues/2295

## Reviewer Checklist:

- [x] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [x] Verify test plan is specified in PR before merging to source

### CODE REVIEW

**Comment by srirambv** [comment]
@petemill @rossmoody could the badge not overlap on shields icon but instead make at the edge? The badge on BAT logo doesn't overlap(at least with single digit) instead is just at the border radius. Could the shields also a similar implementation?

**Comment by rossmoody** [comment]
@srirambv an overlap is pretty standard for notification elements on icons. the shape of the rewards icon just makes it very difficult without appearing completely off balance

**Comment by rossmoody** [comment]
@petemill looks good to me. one note, and i'm unsure if it should be tackled in this pr, but we are removing the `disabled` icon variant (the light grey one). the icons for Rewards and Shields can only ever appear `On` or `Off` and the dropdown panel will provide more context to the user.

**Comment by petemill** [comment]
@srirambv the rewards badge does overlap, but the icon is a triangle so it doesn't look like it does. Please see the mockup at https://github.com/brave/brave-browser/issues/2295 for the coordinates we're aiming for

File: browser/ui/brave_actions/brave_action_icon_with_badge_image_source.cc [hunk 1]
```diff
@@ -19,6 +20,52 @@
 #include "ui/gfx/image/image_skia_operations.h"
 #include "ui/gfx/skia_paint_util.h"
 
+using namespace brave_actions;
+
+void BraveActionIconWithBadgeImageSource::Draw(gfx::Canvas* canvas) {
```
Comment (bridiver) [inline browser/ui/brave_actions/brave_action_icon_with_badge_image_source.cc]: I don't think we want to copy all this code from `IconWithBadgeImageSource::Draw`
That's really the only thing worse than patching. What exactly are we trying to do here?

Comment (bridiver) [reply browser/ui/brave_actions/brave_action_icon_with_badge_image_source.cc]: maybe we can patch in a virtual method and do something like
```
CustomScaleImageSkiaRep(....) ||
skia.AddRepresentation(ScaleImageSkiaRep(...
``` where the default implementation of `CustomScaleImageSkiaRep` just returns false?

Comment (petemill) [reply browser/ui/brave_actions/brave_action_icon_with_badge_image_source.cc]: I'll do something like this. However, it's not just the retrieving of the image, but also the coords of painting due to different alignment requirements.

File: patches/chrome-browser-ui-views-location_bar-location_bar_view.cc.patch [hunk 2]
```diff
@@ -1,5 +1,5 @@
 diff --git a/chrome/browser/ui/views/location_bar/location_bar_view.cc b/chrome/browser/ui/views/location_bar/location_bar_view.cc
-index 16a2d97dab9ffead2178e61cb3b9c65ea8f222c4..4098108819f3f49fbccafb034421628da38ef36b 100644
+index 16a2d97dab9ffead2178e61cb3b9c65ea8f222c4..103eeb6bb91c60c57598cc49e33b1213a2bc61ad 100644
```
Comment (bridiver) [inline patches/chrome-browser-ui-views-location_bar-location_bar_view.cc.patch]: this looks unintentional? No actual patch change?

**Comment by bsclifton** [comment]
Manually ran tests (after noticing check above failed). Everything looks good 👍

**Pull request merged**